lib/jellyfishService.js

```diff
@@ -339,80 +339,84 @@ function createServer(serverConfig, mongoClient, seagullClient, userApiClient, g
-          function(groupId, cb) {
-            async.mapSeries(
-              array,
-              function(obj, cb) {
-                obj._groupId = groupId;
-                dataBroker.addDatum(obj, function(err){
-                  if (err != null) {
-                    if (err.errorCode === 'duplicate') {
-                      duplicates.push(count);
-                      err = null;
-                    } else {
-                      err.dataIndex = count;
+            },
+            function(userId, cb) {
+              getPrivatePair(userId, function(err, privatePair) {
+                cb(err, privatePair ? privatePair.id : null);
+              });
+            },
+            function(groupId, cb) {
+              async.mapSeries(
+                array,
+                function(obj, cb) {
+                  obj._groupId = groupId;
+                  dataBroker.addDatum(obj, function(err){
+                    if (err != null) {
+                      if (err.errorCode === 'duplicate') {
+                        duplicates.push(count);
+                        err = null;
+                      } else {
+                        err.dataIndex = count;
+                      }
                     }
-                  }
-                  ++count;
-                  cb(err);
-                });
-              },
-              cb
-            );
-          }
-        ],
-        function(err) {
-          if (err != null) {
-            if (err.statusCode != null) {
-              // err.message appears to not get serialized if it's an Error, so store it as err.reason
-              err.reason = err.message;
-              res.send(err.statusCode, err);
+                    ++count;
+                    cb(err);
+                  });
+                },
+                cb
+              );
+            }
+          ],
+          function(err) {
+            if (err != null) {
+              if (err.statusCode != null) {
+                // err.message appears to not get serialized if it's an Error, so store it as err.reason
+                err.reason = err.message;
+                res.send(err.statusCode, err);
+              } else {
+                var groupMessage = req.params.groupId ? ('To group[' + req.params.groupId + ']') : '';
+                log.warn(err, 'Problem uploading for user[%s]. %s', userid, groupMessage);
+                res.send(500);
+              }
             } else {
-              var groupMessage = req.params.groupId ? ('To group[' + req.params.groupId + ']') : '';
-              log.warn(err, 'Problem uploading for user[%s]. %s', userid, groupMessage);
-              res.send(500);
+              res.status(200).send(duplicates);
             }
-          } else {
-            res.status(200).send(duplicates);
           }
-        }
-      );
+        );
+      } catch (ex){
+        log.warn(ex, 'an uncaught error occurred!?');
+        res.status(500).send('an uncaught error occurred');
```

Comment:
I'm not sure how the client will handle this, so you might want to check with Jana first, but shouldn't we send back an expected JSON error object?  Something like `{statusCode: 500, message: 'Error getting user private data'}`?